### PR TITLE
Feature/APMI-2442 networking instrumentation flag

### PR DESCRIPTION
**Pull request by Piyush23786**
added isNetworkInstrumentation flag for enabled/disable https spans.

### CODE REVIEW

File: SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift [hunk 1]
```diff
@@ -191,7 +199,9 @@ var splunkRumInitializeCalledTime = Date()
         if options != nil {
             srInit.setAttribute(key: "config_settings", value: options!.toAttributeValue())
         }
-        initalizeNetworkInstrumentation()
+        if options?.isNetworkInstrumentation == true{
```
Comment (johnbley) [inline SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift:202]: `if options?.networkInstrumentation {` - no need to compare with true, that's what `if` does.

File: SplunkRumWorkspace/TestStoryboardApp/TestStoryboardApp/AppDelegate.swift [hunk 2]
```diff
@@ -25,7 +25,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         // Override point for customization after application launch.
-        SplunkRum.initialize(beaconUrl: "http://127.0.0.1:9080/api/v2/spans", rumAuth: "FAKE_RUM_AUTH", options: SplunkRumOptions(allowInsecureBeacon: true, debug: true))
+        SplunkRum.initialize(beaconUrl: "http://127.0.0.1:9080/api/v2/spans", rumAuth: "FAKE_RUM_AUTH", options: SplunkRumOptions(allowInsecureBeacon: true, debug: true,isNetworkInstrumentation: true))
```
Comment (johnbley) [inline SplunkRumWorkspace/TestStoryboardApp/TestStoryboardApp/AppDelegate.swift:28]: Unnecessary change.

**Review by johnbley — CHANGES_REQUESTED** [review]
Please also add a row in the configuration section of `README.md` with this new option.   Thanks!

File: SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift [hunk 3]
```diff
@@ -98,6 +100,12 @@ let SplunkRumVersionString = "0.5.2"
      Enable span creation for screen name changes
      */
     @objc public var screenNameSpans: Bool = true
+    
+    
+    /**
+     Enable NetworkInstrumentation span creation for https calls.
+     */
+    @objc public var isNetworkInstrumentation: Bool = true
```
[the remark this answers is not in the document]
Comment (Piyush23786) [reply SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift]: name changed

File: SplunkRumWorkspace/TestStoryboardApp/TestStoryboardApp/AppDelegate.swift (hunk 2 above)
Comment (Piyush23786) [reply SplunkRumWorkspace/TestStoryboardApp/TestStoryboardApp/AppDelegate.swift]: it's added for testing purposes. will remove that.

File: SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkInstrumentation.swift [hunk 4]
```diff
@@ -199,7 +199,7 @@ extension URLSession {
         return answer
        }
 
-    @objc(splunkSwizzledDownloadTaskWithRequest: completionHandler:) open func splunk_swizzled_downloadTask(with request: URLRequest, completionHandler: @escaping (URL?, URLResponse?, Error?) -> Void) -> URLSessionDownloadTask {
```
[the remark this answers is not in the document]
Comment (Piyush23786) [reply SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkInstrumentation.swift:202]: PR Contains two points.
1. flag to enable/disable Network instrumentation.
2. The second is application crashes in iOS 12 because the networking library based on URLSession does not report HTTP requests (Alamofire SDK).

[the remark this answers is not in the document]
Comment (Piyush23786) [reply SplunkRumWorkspace/SplunkRum/SplunkRum/NetworkInstrumentation.swift:202]: I will create a separate PR for completion handle issues.
yes, I will make the upload task optional.

**Review by johnbley — APPROVED** [review]

**Review by theletterf — APPROVED** [review]
